sensor/common/sensor/central_communication_impl.go

```diff
@@ -242,7 +247,41 @@ func (s *centralCommunicationImpl) initialSync(stream central.SensorService_Comm
 		return err
 	}
 
-	return s.initialPolicySync(stream, detector)
+	if err := s.initialPolicySync(stream, detector); err != nil {
+		return err
+	}
+
+	if s.clientReconciliation {
+		// If client is reconciling, Sensor needs to wait for Central to send the hashes
+		return s.initialDeduperSync(stream, s.syncTimeout)
+	}
+	return nil
+}
+
+func (s *centralCommunicationImpl) initialDeduperSync(stream central.SensorService_CommunicateClient, timeout time.Duration) error {
+	done := make(chan struct{})
+	var err error
+	var msg *central.MsgToSensor
+	go func() {
+		msg, err = stream.Recv()
+		close(done)
+	}()
+	select {
+	case <-time.After(timeout):
```

Comment:
This is great, do you mind opening it in a separate PR, on top of the changes in #7755? There is also a specific test for it: [ROX-17489](https://issues.redhat.com/browse/ROX-17489).

Reply:
Will do